src/identityApiClient.js

```diff
@@ -71,93 +73,109 @@ export default function IdentityAPIClient(mpInstance) {
-                }
-            } catch (e) {
-                mpInstance._Store.identityCallInFlight = false;
-                mpInstance._Helpers.invokeCallback(
-                    callback,
-                    HTTPCodes.noHttpCoverage,
-                    e
-                );
-                mpInstance.Logger.error(
-                    'Error sending identity request to servers with status code ' +
-                        xhr.status +
-                        ' - ' +
-                        e
+        const previousMPID = mpid || null;
+        const uploadUrl = this.getUploadUrl(method, mpid);
+
+        const uploader = window.fetch
+            ? new FetchUploader(uploadUrl)
+            : new XHRUploader(uploadUrl);
+
+        const fetchPayload = {
+            method: 'post',
+            headers: {
+                Accept: 'text/plain;charset=UTF-8',
+                'Content-Type': 'application/json',
+                'x-mp-key': mpInstance._Store.devToken,
+            },
+            body: JSON.stringify(identityApiRequest),
+        };
+
+        try {
+            mpInstance._Store.identityCallInFlight = true;
+            const response = await uploader.upload(fetchPayload);
+
+            let identityResponse;
+
+            if (response.json) {
+                // https://go.mparticle.com/work/SQDSDKS-6568
+                // FetchUploader returns the response as a JSON object that we have to await
+                const responseBody = await response.json();
+                identityResponse = this.getIdentityResponseFromFetch(
+                    response,
+                    responseBody
                 );
+            } else {
+                identityResponse = this.getIdentityResponseFromXHR(response);
             }
+
+            verbose(
+                'Received Identity Response from server: ' +
+                    JSON.stringify(identityResponse.responseText)
+            );
+
+            parseIdentityResponse(
+                identityResponse,
+                previousMPID,
+                callback,
+                originalIdentityApiData,
+                method,
+                knownIdentities,
+                false
+            );
+        } catch (err) {
+            mpInstance._Store.identityCallInFlight = false;
+            invokeCallback(callback, HTTPCodes.noHttpCoverage, err);
+            error('Error sending identity request to servers' + ' - ' + err);
         }
     };
+
+    this.getUploadUrl = (method, mpid) => {
+        const uploadServiceUrl = mpInstance._Helpers.createServiceUrl(
+            mpInstance._Store.SDKConfig.identityUrl
+        );
+
+        const uploadUrl =
+            method === Modify
+                ? uploadServiceUrl + mpid + '/' + method
+                : uploadServiceUrl + method;
+
+        return uploadUrl;
+    };
+
+    this.getIdentityResponseFromFetch = (response, responseBody) => ({
```

Comment:
Note to self: Make this a private method

Reply:
Is this still relevant?

Reply:
Not at this time. I think I'll do this when we migrate to TS.